code/datums/inventory.dm

```diff
@@ -0,0 +1,112 @@
+#define MAKE_SLOT(slot, item) list("id" = slot, "item" = item?.name)
+
+/datum/humanInventory
+	var/mob/living/carbon/human/human = null
+
+/datum/humanInventory/New(var/mob/living/carbon/human/H)
+	..()
+	human = H
+
+/datum/humanInventory/ui_state(mob/user)
+	return tgui_physical_state
+
+/datum/humanInventory/ui_status(mob/user)
+	return min(
+		tgui_physical_state.can_use_topic(src.human, user),
+		tgui_not_incapacitated_state.can_use_topic(src.human, user)
+	)
+
+/datum/humanInventory/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "HumanInventory")
+		ui.open()
+
+/datum/humanInventory/ui_data(mob/user)
+
+	var/list/slots = list(
+		MAKE_SLOT("slot_head", src.human.head),
+		MAKE_SLOT("slot_wear_mask", src.human.wear_mask),
+		MAKE_SLOT("slot_glasses", src.human.glasses),
+		MAKE_SLOT("slot_ears", src.human.ears),
+		MAKE_SLOT("slot_l_hand", src.human.l_hand),
+		MAKE_SLOT("slot_r_hand", src.human.r_hand),
+		MAKE_SLOT("slot_gloves", src.human.gloves),
+		MAKE_SLOT("slot_shoes", src.human.shoes),
+		MAKE_SLOT("slot_belt", src.human.belt),
+		MAKE_SLOT("slot_w_uniform", src.human.w_uniform),
+		MAKE_SLOT("slot_wear_suit", src.human.wear_suit),
+		MAKE_SLOT("slot_back", src.human.back),
+		MAKE_SLOT("slot_wear_id", src.human.wear_id),
+		MAKE_SLOT("slot_l_store", src.human.l_store),
+		MAKE_SLOT("slot_r_store", src.human.r_store),
+	)
+
+	. = list(
+		"name" = src.human.name,
+		"slots" = slots,
+		"handcuffed" = src.human.hasStatus("handcuffed"),
+		"internal" = src.human.internal != null,
+		"canSetInternal" = istype(src.human.wear_mask, /obj/item/clothing/mask) && istype(src.human.back, /obj/item/tank) && !src.human.internal
+	)
```

Comment:
if `update_inv()` exists why can't this all be static and it be refreshed by that

Reply:
I dont think im going to be able to find all the possible places where they might get modified. I can do a pass over it later, but I'd rather keep this to interface changes for now.

I also don't want accidental desyncs between what's in their inventory and what you can see in the interface.

Reply:
fair